### PR TITLE
Reflect pylint suggestions

### PR DESCRIPTION
**Pull request by panpilkarz**
https://github.com/beamer-bridge/beamer/issues/1812

### CODE REVIEW

File: scripts/deployment/main.py [hunk 1]
```diff
@@ -58,11 +58,11 @@ def get_commit_id() -> str:
...
-        name: dict(
-            address=contract.address,
-            deployment_block=contract.deployment_block,
-            deployment_args=contract.deployment_args,
-        )
+        name: {
+            "address": contract.address,
+            "deployment_block": contract.deployment_block,
+            "deployment_args": contract.deployment_args,
+        }
```
Comment (istankovic) [inline scripts/deployment/main.py:65]: Please drop this change, this is a case where the dict keywords actually make more sense, ergonomically speaking.

**Comment by istankovic** [comment]
Please drop commit `120f3a2b57433a4d0e86a9d4381c2692f4eeed4a`.

File: scripts/diff_gas_profiles.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ class Profile:
 def _read_profile(filename: str) -> Profile:
     contract = None
     profile = Profile(filename=filename)
-    with open(filename) as fp:
+    with open(filename, encoding="utf-8") as fp:
```
Comment (istankovic) [inline scripts/diff_gas_profiles.py]: Why adding this change and then removing this later in the series?
If this hunk is not meant to be in the PR at all, then it just adds noise and should be removed.

File: beamer/agent/models/claim.py [hunk 3]
```diff
@@ -128,8 +128,8 @@ def get_minimum_challenge_stake(self, initial_claim_stake: Wei) -> Wei:
 
         if challenger_stake > claimer_stake:
             return Wei(challenger_stake - claimer_stake + initial_claim_stake)
-        else:
-            return Wei(claimer_stake - challenger_stake + 1)
+
+        return Wei(claimer_stake - challenger_stake + 1)
```
Comment (istankovic) [inline beamer/agent/models/claim.py:132]: Ah, nice. :+1:

File: beamer/agent/state_machine.py [hunk 4]
```diff
@@ -85,35 +85,34 @@ def process_event(event: Event, context: Context) -> HandlerResult:
...
+    if isinstance(event, FillInvalidatedResolved):
         return _handle_fill_invalidated_resolved(event, context)
 
-    elif isinstance(event, ChainUpdated):
+    if isinstance(event, ChainUpdated):
         return _handle_chain_updated(event, context)
 
-    else:
-        raise RuntimeError("Unrecognized event type")
+    raise RuntimeError("Unrecognized event type")
```
Comment (istankovic) [inline beamer/agent/state_machine.py:115]: Ugh, I see where this is coming from but it's... meh.
I'm thinking we should rather convert the entire thing to a `match` statement.

File: beamer/middleware.py [hunk 5]
```diff
@@ -226,50 +226,49 @@ def _rate_limiter_inner(
...
+        rate_limited, response = _try_make_request(make_request, method, params)
+        if rate_limited:
+            log.debug(
+                "Entering rate limiting mode", thread=threading.current_thread().name, rpc=rpc
+            )
+            state.rate_limit_end = time.time() + _RATE_LIMIT_PERIOD
+            time.sleep(_RATE_LIMIT_REQUEST_DELAY)
+        else:
+            assert response is not None
+            return response
```
Comment (istankovic) [inline beamer/middleware.py:271]: I have no idea what happened here and github at least is not helpful when showing the diff.
This is some of the trickiest code in our agent, could you please explain what you did here?

File: beamer/util.py [hunk 6]
```diff
@@ -183,7 +183,7 @@ def make_web3(
 def _load_ERC20_abi() -> tuple[Any, ...]:
     path = pathlib.Path(__file__)
     path = path.parent.joinpath("data/abi/StandardToken.json")
-    with path.open("rt") as fp:
+    with path.open("rt", encoding="utf-8") as fp:
```
Comment (istankovic) [inline beamer/util.py]: Why add this here, only to remove in a later commit?
Could you please remove such unnecessary hunks?

File: beamer/tests/agent/multiple_processors/test_multiple_event_processors.py [hunk 7]
```diff
@@ -119,7 +119,7 @@ def _get_config(
     deployment_info = {}
 
     for chain_id in chain_map:
-        contracts_info = dict{
+        contracts_info = {
```
Comment (istankovic) [inline beamer/tests/agent/multiple_processors/test_multiple_event_processors.py:122]: Well, that's one of the strangest hunks I've seen. What happened here? :-)

File: beamer/tests/agent/test_rate_limiter.py [hunk 8]
```diff
@@ -65,5 +65,5 @@ def test_rate_limiting_rpc(config):
         "log_level": "debug",
         "rpc": proxy_l2a.url(),
         "thread": "MainThread",
-    )
+    }
```
Comment (istankovic) [inline beamer/tests/agent/test_rate_limiter.py:68]: How was this not a syntax error before?

**Review by istankovic — CHANGES_REQUESTED** [review]
I think this a move in the right direction when it comes to lints, but this needs more work.

**Comment by panpilkarz** [comment]
Thanks for the feedback @istankovic 
Going to address all items today.

File: beamer/middleware.py (hunk 5 above)
Comment (panpilkarz) [reply beamer/middleware.py:271]: No need to do `else` after `return`

**Comment by panpilkarz** [comment]
> Please drop commit 120f3a2

out of curiosity: why?

File: .pylintrc [hunk 9]
```diff
@@ -32,12 +32,22 @@ disable=
...
     redefined-outer-name,
+    global-statement,
+    unreachable,
+    fixme,
+    broad-exception-caught,
+    too-few-public-methods,
+    super-init-not-called,
+    c-extension-no-member,
+    redefined-argument-from-local,
+    attribute-defined-outside-init,
```
[the remark this answers is not in the document]
Comment (panpilkarz) [reply .pylintrc:53]: Four:
```
************* Module beamer.tests.util
beamer/tests/util.py:53:8: W0201: Attribute '_end' defined outside __init__ (attribute-defined-outside-init)
beamer/tests/util.py:105:8: W0201: Attribute '_thread' defined outside __init__ (attribute-defined-outside-init)
************* Module beamer.agent.chain
beamer/agent/chain.py:76:8: W0201: Attribute '_thread' defined outside __init__ (attribute-defined-outside-init)
beamer/agent/chain.py:180:8: W0201: Attribute '_thread' defined outside __init__ (attribute-defined-outside-init)
```

[the remark this answers is not in the document]
Comment (panpilkarz) [reply .pylintrc:53]: Easy to fix by adding `self._thread = None` in `__init__()`

**Comment by istankovic** [comment]
> > Please drop commit [120f3a2](https://github.com/beamer-bridge/beamer/commit/120f3a2b57433a4d0e86a9d4381c2692f4eeed4a)
> 
> out of curiosity: why?
> 
> Were all `dict()` intentional and should I mark them in code with `# pylint: disable=use-dict-literal` or disable it totally in `.pylintrc`?

First off, sorry I was so terse here, I should have definitely provided more context.

Yes, pretty much all the `dict()` instances were intentional and e.g. if you take a look at 
```python
        params = dict(fromBlock=self._from_block, toBlock=to_block, address=self._address)
```
vs.
```python
        params = {
            "fromBlock": self._from_block,
            "toBlock": to_block,
            "address": self._address,
        }
```
the lint just forces you to use the more verbose, less ergonomic option. Less ergonomic because one needs to consider the quotes.

That is not to say that we never use dict literals, I do think there are examples of that in the code. It's just that in these
cases we opted for what seemed to be "better".
 
As is the case very often in programming, it is a matter of taste. In general, we follow the generally accepted Python conventions (especially e.g. PEP8), but also reconsider the convention if it makes our code harder to read, less ergonomic, more complex etc. I think I would phrase it as "generally, the linter is quite helpful, but if it makes your code harder to read and maintain, the linter is wrong".

File: .pylintrc (hunk 9 above)
[the remark this answers is not in the document]
Comment (panpilkarz) [reply .pylintrc]: Yup, mypy was complaining and I couldn't work it around. Finally disabled these warnings in the code.

File: beamer/tests/util.py [hunk 10]
```diff
@@ -144,7 +147,11 @@ def _fetch_events(self) -> None:
         to_block = ape.chain.blocks.height
         if to_block < self._from_block:
             return
-        params = dict(fromBlock=self._from_block, toBlock=to_block, address=self._address)
+        params = dict(
+            fromBlock=self._from_block,
+            toBlock=to_block,
+            address=self._address,
+        )
```
Comment (istankovic) [inline beamer/tests/util.py:154]: Why was the formatting changed here?

File: beamer/events.py [hunk 11]
```diff
@@ -262,7 +262,9 @@ def _fetch_range(
         try:
             before_query = time.monotonic()
             params: FilterParams = dict(
-                fromBlock=from_block, toBlock=to_block, address=self._contract_addresses
+                fromBlock=from_block,
+                toBlock=to_block,
+                address=self._contract_addresses,
```
Comment (istankovic) [inline beamer/events.py:267]: Why was the formatting changed here?

**Review by istankovic — APPROVED** [review]
Great work, thanks!

**Pull request merged**